File: app/schemas/category.py
from pydantic import BaseModel
from datetime import datetime

class CategoryCreate(BaseModel):
    name: str
    type: str  # "income" or "expense"

class CategoryUpdate(BaseModel):
    name: str | None = None
    type: str | None = None

class CategoryResponse(BaseModel):
    id: int
    name: str
    type: str
    is_default: bool
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/category.ts
// Mirrors app/schemas/category.py.

export interface CategoryResponse {
  id: number;
  name: string;
  type: string; // "income" | "expense" by convention, not enforced by the backend schema
  is_default: boolean;
  created_at: string;
}

export interface CategoryCreatePayload {
  name: string;
  type: string;
}

export interface CategoryUpdatePayload {
  name?: string;
  type?: string;
}
